File: src/main/java/pl/patrykdolata/lifemanager/security/AuthenticatedUser.java
package pl.patrykdolata.lifemanager.security;

import lombok.EqualsAndHashCode;
import lombok.Value;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.Collection;

@Value
@EqualsAndHashCode(callSuper = true)
public class AuthenticatedUser extends User {

    Long id;

    public AuthenticatedUser(Long id, String username, String password, boolean activated,
                             Collection<GrantedAuthority> authorities) {
        super(username, password, activated, true, true, true,
                authorities);
        this.id = id;
    }
}

File: src/main/kotlin/pl/patrykdolata/lifemanager/security/AuthenticatedUser.kt
package pl.patrykdolata.lifemanager.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User

class AuthenticatedUser(
        val id: Long, username: String, password: String = "", val email: String = "", val firstName: String = "",
        val lastName: String = "", activated: Boolean = true, authorities: Collection<GrantedAuthority> = emptySet()
) : User(username, password, activated, true, true, true, authorities)
